store/mongo/datastore_mongo_test.go

```diff
@@ -1254,12 +1261,18 @@ func TestStoreGetDevices(t *testing.T) {
 				}
 			}
 			if tc.filter.IDs != nil {
-				if assert.Len(t, dbdevs, len(tc.filter.IDs)) {
-					for i := range dbdevs {
-						assert.Equal(t,
-							dbdevs[i].Id,
-							tc.filter.IDs[i])
-					}
+				deviceIDs := make(map[string]struct{}, len(tc.filter.IDs))
+				for _, id := range tc.filter.IDs {
+					deviceIDs[id] = struct{}{}
+				}
+				var lastStatus string
```

Comment:
where is this variable assigned a value?

Reply:
Nice catch! It was supposed to be assigned the status of the current element at the end of the loop.